src/parts/Quests.svelte

```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+  import type { WebfishingSave } from "../game/types";
+  import QuestItem from "../components/QuestItem.svelte";
+  import Section from "../components/Section.svelte";
+  import type { GodotCustomDictionary, GodotCustomArray, GodotString, GodotInt, GodotReal, GodotBool } from "../lib/types";
+
+  export let save: WebfishingSave;
+
+  // Extract quests, ensuring nested value types are handled properly
+  let quests = save.value.quests.value;
+  const questKeys = Object.keys(quests).filter((key) => !key.startsWith("_"));
+
+  // Helper function to safely access title and other fields in a quest
+  const getQuestTitle = (questKey: string) => {
+    return quests[questKey]?.value?.title?.value || "Untitled Quest";
+  };
+
+  const getQuest = (questKey: string) => {
+    const quest = quests[questKey]?.value;
+
+    return {
```

Comment:
I'm not a huge fan of doing it this way (I think it makes more sense to modify the actual Godot variant wrappers like how the other components do it) but not that important.